packages/headless/src/features/configuration/analytics-params.ts

```diff
@@ -1,38 +1,36 @@
+import {Relay} from '@coveo/relay';
 import {EventDescription} from 'coveo.analytics';
 import {getAnalyticsSource} from '../../api/analytics/analytics-selectors';
-import {getVisitorID} from '../../api/analytics/coveo-analytics-utils';
 import {getPageID} from '../../api/analytics/search-analytics';
 import {AnalyticsParam} from '../../api/search/search-api-params';
+import {NavigatorContext} from '../../app/navigatorContextProvider';
 import {AnalyticsState} from './configuration-state';
 
-export const fromAnalyticsStateToAnalyticsParams = async (
+export const fromAnalyticsStateToAnalyticsParams = (
   s: AnalyticsState,
+  navigatorContext: NavigatorContext,
+  relay: Relay,
   eventDescription?: EventDescription
-): Promise<AnalyticsParam> => {
-  const isNextAnalytics = s.analyticsMode === 'next';
-  const isDefaultReferrer = s.originLevel3 === 'default';
+): AnalyticsParam => {
   return {
     analytics: {
-      clientId: await getVisitorID(s),
+      clientId: relay.getMeta('').clientId,
       clientTimestamp: new Date().toISOString(),
-      documentReferrer:
-        isNextAnalytics && isDefaultReferrer ? null : s.originLevel3,
+      documentReferrer: navigatorContext.referrer,
```

Comment:
I ensured that we always send null where we sent default, and that's all.

Further improvement could probably be made, but I do prefer to close this PR, it has lived long enough.

FYI @samisayegh probably some potential hiccup with documentLocation similar to #3974
No regression tho, so I will not hold this PR because of that.

Reply:
Since the initial PR, we've made referrer [less strict](https://github.com/coveo-platform/analytics_schema/pull/435) (i.e., `urlEmpty`), so the allowed types now are:

```
location: uri | null
referrer: uri | "" | null
```
We didn't support an empty string for `location` because unlike `document.referrer`, we didn't see `window.location.href` return empty string, even on a new tab.

When navigatorContext is in nodejs mode, would it be possible to send `null` for both keys?
Also if it's helpful, `relay.getMeta()` has this behavior encoded into it. That said, getting access to the relay instance might not be so easy,